docs/architecture.md

```diff
@@ -0,0 +1,292 @@
+# All About `armada-spark`
+
+## Running `armada-spark`
+
+`armada-spark`'s main entry point is a class called [`ArmadaSparkSubmit`](https://github.com/ClifHouck/armada-spark/blob/master/src/main/scala-spark-4.1/org/apache/spark/deploy/ArmadaSparkSubmit.scala).
+It's designed to be a drop-in replacement/extension for `spark-submit`.
+However, due to limitations in upstream integration, we load it via the 
```

Comment:
> due to limitations in upstream integration

That phrase seems a bit awkward.  Would it be better to say something like:
"and we are working on getting it accepted upstream: https://github.com/apache/spark/pull/50770

Reply:
Addressed in 14596ef